Core/include/Acts/Seeding/SpacePointGrid.ipp

```diff
@@ -10,57 +10,78 @@
 
+#include <iostream>
+
+using AxisScalar = Acts::Vector3::Scalar;
+
 template <typename SpacePoint>
 std::unique_ptr<Acts::SpacePointGrid<SpacePoint>>
 Acts::SpacePointGridCreator::createGrid(
     const Acts::SpacePointGridConfig& config) {
-  int phiBins;
-  // for no magnetic field, create 100 phi-bins
-  if (config.bFieldInZ == 0) {
-    phiBins = 100;
-  } else {
-    // calculate circle intersections of helix and max detector radius
-    float minHelixRadius =
-        config.minPt / (300. * std::abs(config.bFieldInZ));  // in mm
-    float maxR2 = config.rMax * config.rMax;
-    float xOuter = maxR2 / (2 * minHelixRadius);
-    float yOuter = std::sqrt(maxR2 - xOuter * xOuter);
-    float outerAngle = std::atan(xOuter / yOuter);
-    // intersection of helix and max detector radius minus maximum R distance
-    // from middle SP to top SP
-    float innerAngle = 0;
-    if (config.rMax > config.deltaRMax) {
-      float innerCircleR2 =
-          (config.rMax - config.deltaRMax) * (config.rMax - config.deltaRMax);
-      float xInner = innerCircleR2 / (2 * minHelixRadius);
-      float yInner = std::sqrt(innerCircleR2 - xInner * xInner);
-      innerAngle = std::atan(xInner / yInner);
-    }
-
-    // FIXME: phibin size must include max impact parameters
-    // divide 2pi by angle delta to get number of phi-bins
-    // size is always 2pi even for regions of interest
-    phiBins = std::floor(2 * M_PI / (outerAngle - innerAngle));
+ 
+  // calculate circle intersections of helix and max detector radius
+  float minHelixRadius = config.minPt / (300. * config.bFieldInZ);  // in mm -> R[mm] =pT[GeV] / (3·10−4×B[T]) =  pT[MeV] / (300 *Bz[kT])
+  float maxR2 = config.rMax * config.rMax;
+  float xOuter = maxR2 / (2 * minHelixRadius);
+  float yOuter = std::sqrt(maxR2 - xOuter * xOuter);
+  float outerAngle = std::atan(xOuter / yOuter);
+  // intersection of helix and max detector radius minus maximum R distance from
+  // middle SP to top SP
+  float innerAngle = 0;
+  float Rmin = config.rMax;
+  if (config.rMax > config.deltaRMax) {
+    Rmin = config.rMax - config.deltaRMax;
+    float innerCircleR2 =
+        (config.rMax - config.deltaRMax) * (config.rMax - config.deltaRMax);
+    float xInner = innerCircleR2 / (2 * minHelixRadius);
+    float yInner = std::sqrt(innerCircleR2 - xInner * xInner);
+    innerAngle = std::atan(xInner / yInner);
   }
+
+  // evaluating the azimutal deflection including the maximum impact parameter
+  float deltaAngleWithMaxD0 = std::abs(std::asin(config.impactMax/(Rmin)) - std::asin(config.impactMax/config.rMax));
+  
+  // evaluating delta Phi based on the inner and outer angle, and the azimutal deflection including the maximum impact parameter
+  float deltaPhi = (outerAngle - innerAngle + deltaAngleWithMaxD0)/float(config.numberOfPhiBins);
+  
+  // divide 2pi by angle delta to get number of phi-bins
+  // size is always 2pi even for regions of interest
+  int phiBins = std::ceil(2 * M_PI / deltaPhi);
```

Comment:
Do I read this correctly that previously, `phiBins` was **always** 1? (@robertlangenberg)

Reply:
certainly shouldn't have been the case, did you spot a bug?
also, I used std::floor in order to err on the side of having larger bins, to be sure to catch all relevant SP.